### PR TITLE
feat(pie-docs): DSW-1485 add switch code page

### PR DESCRIPTION
**Pull request by fernandofranca**
## Describe your changes (can list changeset entries if preferable)

- [Added] - Switch component code page

## Author Checklist (complete before requesting a review)
- [X] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [X] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/components/switch/overview/overview.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    key: Overview
+    parent: Switch
+    order: 1
+shouldShowContents: true
+---
+
+## Overview
+The purpose of switches is to offer users a quick and straightforward method to switch between two distinct options or states. Switches enhance the user experience by providing a clear and efficient way to make selections or enable/disable functionality.
+
+Switches are commonly used to enable or disable settings, switch between modes or perform binary actions in user interfaces. When users interact with switches, they change their appearance to indicate the current state or selection.
```
Comment (webbertakken) [inline apps/pie-docs/src/components/switch/overview/overview.md:12]: Perhaps a recommendation of what to use when there are more than 2 states (delivery/pickup/dine-in could be one example)

File: apps/pie-docs/src/components/switch/overview/overview.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+---
+
+## Examples
+
+Outlines the atomic level interactive elements for the component.
+
+### LTR examples
+
+Here are some examples of switches in left to right context:
```
Comment (webbertakken) [inline apps/pie-docs/src/components/switch/overview/overview.md:206]:
```suggestion
Here are some examples of switches in left-to-right context:
```

**Review by webbertakken — APPROVED** [review]
Great!

File: apps/pie-docs/src/components/switch/overview/overview.md (hunk 1 above)
Comment (fernandofranca) [reply apps/pie-docs/src/components/switch/overview/overview.md:12]: Added a comment to the ticket, so Designers can check it during Design review.

File: apps/pie-docs/src/components/switch/code/code.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+<pie-switch isChecked label="Label" labelPlacement="trailing" @pie-switch-changed="handleChange"></pie-switch>
+```
+
+For React Applications.
+
+```js
+// React templates (using Next 13)
+import { PieSwitch } from '@justeattakeaway/pie-switch/dist/react';
+
+<pie-switch isChecked label="Label" labelPlacement="trailing" onSwitchChangedEvent={handleChange}></pie-switch>
```
Comment (raoufswe) [inline apps/pie-docs/src/components/switch/code/code.md:98]: the event should be `onPieSwitchChanged`
```suggestion
<pie-switch isChecked label="Label" labelPlacement="trailing" onPieSwitchChanged={handleChange}></pie-switch>
```
Screenshot 2023-11-20 at 09 59 17

**Review by maledr5 — APPROVED** [review]

**Pull request merged**